### PR TITLE
[ADD] Pdfplumber support

### PR DESCRIPTION
**Pull request by bosd**
This is a highly opinionated basic implementation of pdfplumber.
I've created this in an attempt to parse invoices with an html style table as mentioned in #359 

It's working, but currently I'm developing an integration with camelot.
As Camelot allows more complex table structures to be parsed and a gui to help build the templates.

Creating this PR in case someone needs it.

### CODE REVIEW

**Comment by bosd** [comment]
Propably need to tell the tests to install pdfplumber.
@m3nu Any ideas how to do that?

**Comment by bosd** [comment]
All ✔️ now!
But in this state it is not something we would like to merge.
Need to clean commits and dependency issue on pdfminer.
Therefore will convert this one back to draft.

**Comment by bosd** [comment]
depends upon: https://github.com/invoice-x/invoice2data/pull/395

File: README.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ Choose any of the following input readers:
 - pdftotext `invoice2data --input-reader pdftotext invoice.pdf`
 - tesseract `invoice2data --input-reader tesseract invoice.pdf`
 - pdfminer.six `invoice2data --input-reader pdfminer invoice.pdf`
+- pdf plumber `invoice2data --input-reader pdfplumber invoice.pdf`
```
Comment (rmilecki) [inline README.md:59]: Nitpicking: I'd replace `pdf plumber` with `pdfplumber` for consistency.

**Comment by rmilecki** [comment]
Looks good except for the list of commits in your `bosd:pdfplumber` branch. There is some merge commit and a lot of work-in-progress commits.

In my personal taste we **shouldn't** merge such commits into `invoice2data`' `master` branch. Those commits don't bring much/any value, make `git log` a bit messy and make even break `git bisect` if someone ever tries that.

My personal suggestion / preference would be to squash all those commits into a single one with some clear subject.

**Comment by bosd** [comment]
> In my personal taste we **shouldn't** merge such commits into `invoice2data`' `master` branch.

@rmilecki  
I totally agree. Was really struggling on this one. (working on 2 dev systems one windows and one linux machine)
Will try to fiup here. otherwise start over on new branch and pr.

**Pull request closed without merging**